src/parser.rs

```diff
@@ -2044,17 +2046,223 @@ impl<'a> Parser<'a> {
+                }
+                Keyword::CREATEDB | Keyword::NOCREATEDB => {
+                    if create_db.is_some() {
+                        parser_err!("Found multiple CREATEDB or NOCREATEDB")
+                    } else {
+                        create_db = Some(keyword == Keyword::CREATEDB);
+                        Ok(())
+                    }
+                }
+                Keyword::CREATEROLE | Keyword::NOCREATEROLE => {
+                    if create_role.is_some() {
+                        parser_err!("Found multiple CREATEROLE or NOCREATEROLE")
+                    } else {
+                        create_role = Some(keyword == Keyword::CREATEROLE);
+                        Ok(())
+                    }
+                }
+                Keyword::SUPERUSER | Keyword::NOSUPERUSER => {
+                    if superuser.is_some() {
+                        parser_err!("Found multiple SUPERUSER or NOSUPERUSER")
+                    } else {
+                        superuser = Some(keyword == Keyword::SUPERUSER);
+                        Ok(())
+                    }
+                }
+                Keyword::REPLICATION | Keyword::NOREPLICATION => {
+                    if replication.is_some() {
+                        parser_err!("Found multiple REPLICATION or NOREPLICATION")
+                    } else {
+                        replication = Some(keyword == Keyword::REPLICATION);
+                        Ok(())
+                    }
+                }
+                Keyword::PASSWORD => {
+                    if password.is_some() {
+                        parser_err!("Found multiple PASSWORD")
+                    } else {
+                        password = if self.parse_keyword(Keyword::NULL) {
+                            Some(Password::NullPassword)
+                        } else {
+                            Some(Password::Password(Expr::Value(self.parse_value()?)))
+                        };
+                        Ok(())
+                    }
+                }
+                Keyword::CONNECTION => {
+                    self.expect_keyword(Keyword::LIMIT)?;
+                    if connection_limit.is_some() {
+                        parser_err!("Found multiple CONNECTION LIMIT")
+                    } else {
+                        connection_limit = Some(Expr::Value(self.parse_number_value()?));
+                        Ok(())
+                    }
+                }
+                Keyword::VALID => {
+                    self.expect_keyword(Keyword::UNTIL)?;
+                    if valid_until.is_some() {
+                        parser_err!("Found multiple VALID UNTIL")
+                    } else {
+                        valid_until = Some(Expr::Value(self.parse_value()?));
+                        Ok(())
+                    }
+                }
+                Keyword::IN => {
+                    if self.parse_keyword(Keyword::ROLE) || self.parse_keyword(Keyword::GROUP) {
```

Comment:
> IN GROUP is an obsolete spelling of IN ROLE.

(postgres docs)

Reply:
I think the typical pattern in `sqlparser` is that we parse the sql as provided into an AST that could recover the same pattern rather than applying semantic meaning

So in this case, can you please change `CreateRole` to have an explicit `group: Option<Ident>` rather than reusing `in_role`? 

The same comment applies to `Role/User`

Reply:
Here is my proposal: https://github.com/sqlparser-rs/sqlparser-rs/pull/628